File: pdlearning/src/microservices/course-microservice/Api/Application/Models/EmailTemplateModel.cs
using System;
using System.Collections.Generic;

namespace Microservice.Course.Application.Models
{
    public class EmailTemplateModel
    {
        public Guid Id { get; set; }

        public string Template { get; set; }

        public Dictionary<string, string> Tags { get; set; } = new Dictionary<string, string>();
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/email-template.model.ts
export interface IEmailTemplateModel {
  id: string;
  template: string;
  tags?: Dictionary<string>;
}

export class EmailTemplateModel implements IEmailTemplateModel {
  public id: string = '';
  public template: string = '';
  public tags: Dictionary<string> = {};

  constructor(data?: IEmailTemplateModel) {
    if (data == null) {
      return;
    }
    this.id = data.id;
    this.template = data.template;
    this.tags = data.tags ? data.tags : {};
  }
}
